library/cwm/src/lib/cwm/dialog.rb

```diff
@@ -0,0 +1,97 @@
+require "yast"
+
+Yast.import "CWM"
+
+module CWM
+  # An OOP API and the pieces missing from {YastClass::CWM#show Yast::CWM.show}:
+  # - creating and closing a wizard dialog
+  # - Back/Abort/Next buttons
+  #
+  # @see UI::Dialog
+  # @see CWM::AbstractWidget
+  class Dialog
+    include Yast::Logger
+    include Yast::I18n
+    include Yast::UIShortcuts
+
+    # @return [String,nil] Set a title, or keep the existing title
+    abstract_method :title
+
+    # @return [CWM::WidgetTerm]
+    abstract_method :contents
+
+    # A shortcut for `.new(*args).run`
+    def self.run(*args)
+      new(*args).run
+    end
+
+    # The entry point.
+    # Will open (and close) a wizard dialog unless one already exists.
+    # @return [Symbol]
+    def run
+      if should_open_dialog?
+        wizard_create_dialog { run_assuming_open }
+      else
+        run_assuming_open
+      end
+    end
+
+    def wizard_create_dialog(&block)
+      Yast::Wizard.CreateDialog
+      block.call
+    ensure
+      Yast::Wizard.CloseDialog
+    end
+
+    def run_assuming_open
```

Comment:
I really do not like this name. Also why it is public method? and why it missing documentation if it is intention to be a public method?

I suggest something like `def show`